packages/opentelemetry-instrumentation-haystack/opentelemetry/instrumentation/haystack/utils.py

```diff
@@ -1,8 +1,69 @@
+import dataclasses
+import json
 import logging
+import os
 import traceback
+
+from opentelemetry import context as context_api
 from opentelemetry.instrumentation.haystack.config import Config
 
 
+class EnhancedJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if dataclasses.is_dataclass(o):
+            return dataclasses.asdict(o)
+        if hasattr(o, "to_json"):
+            return o.to_json()
+        return super().default(o)
+
+
+def should_send_prompts():
+    return (
+        os.getenv("TRACELOOP_TRACE_CONTENT") or "true"
+    ).lower() == "true" or context_api.get_value("override_enable_content_tracing")
+
+
+def dont_throw(func):
+    """
+    A decorator that wraps the passed in function and logs exceptions instead of throwing them.
+
+    @param func: The function to wrap
+    @return: The wrapper function
+    """
+    # Obtain a logger specific to the function's module
+    logger = logging.getLogger(func.__module__)
+
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            logger.debug(
+                "OpenLLMetry failed to trace in %s, error: %s", func.__name__, str(e)
+            )
+            if Config.exception_logger:
+                Config.exception_logger(e)
+
+    return wrapper
+
+
+@dont_throw
+def process_request(args, kwargs):
+    kwargs_to_serialize = kwargs.copy()
+    for arg in args:
+        if arg and isinstance(arg, dict):
+            for key, value in arg.items():
+                kwargs_to_serialize[key] = value
+    args_to_serialize = [arg for arg in args if not isinstance(arg, dict)]
+    input_entity = {"args": args_to_serialize, "kwargs": kwargs_to_serialize}
+    return json.dumps(input_entity, cls=EnhancedJSONEncoder)
```

Comment:
@tibor-reiss almost missed this - should be behind `if should_send_prompts()`

Reply:
Good catch, thanks!

Reply:
@nirga Just looked at it a second time: should this check rather be in wrap_pipeline.py? Otherwise both spans will be set to None.
I was using the implementation in langchain as blueprint, however I did not like that much that the process_request / process_response functions change state (it's kinda a side effect, and also harder to test).
Let me know your preference!